Mathlib/LinearAlgebra/Matrix/PosDef.lean

```diff
@@ -165,4 +166,25 @@ def InnerProductSpace.ofMatrix {M : Matrix n n 𝕜} (hM : M.PosDef) :
   InnerProductSpace.ofCore _
 #align matrix.inner_product_space.of_matrix Matrix.InnerProductSpace.ofMatrix
 
+section ConjTransposeMul
+
+variable {m : Type _} [Fintype m]
+open BigOperators
+
+/-- The conjugate transpose of a matrix mulitplied by the matrix is positive semidefinite -/
+theorem isPosSemidef_conjTranspose_mul_self (A : Matrix m n 𝕜) : Matrix.PosSemidef (Aᴴ ⬝ A) := by
+  refine ⟨ isHermitian_transpose_mul_self _, fun x => ?_ ⟩
+  rw [← mulVec_mulVec, dotProduct_mulVec, vecMul_conjTranspose, star_star, dotProduct,
+    IsROrC.re_sum]
+  simpa [Pi.star_apply, IsROrC.star_def, IsROrC.mul_re, IsROrC.conj_re, IsROrC.conj_im, neg_mul,
+    sub_neg_eq_add] using
+  Finset.sum_nonneg (fun i => (fun _ => add_nonneg (mul_self_nonneg _) (mul_self_nonneg _)))
+
+/-- A matrix multiplied by its conjugate transpose is positive semidefinite -/
+theorem isPosSemidef_self_mul_conjTranspose (A : Matrix m n 𝕜) : Matrix.PosSemidef (A ⬝ Aᴴ) := by
+  nth_rewrite 1 [← conjTranspose_conjTranspose A]
+  apply isPosSemidef_conjTranspose_mul_self
```

Comment:
Does this work?
```suggestion
theorem isPosSemidef_self_mul_conjTranspose (A : Matrix m n 𝕜) : Matrix.PosSemidef (A ⬝ Aᴴ) :=
  isPosSemidef_conjTranspose_mul_self Aᴴ
```

Reply:
No

Reply:
I changed it to `simpa [conjTranspose_conjTranspose] using isPosSemidef_conjTranspose_mul_self Aᴴ`